test/refactoring/ConvertAsync/convert_to_continuation.swift

```diff
@@ -341,6 +448,100 @@ func testResultFromValueAndError(completionHandler: (Result<Int, Error>) -> Void
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:       if let error = error {
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:         continuation.resume(throwing: error)
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:       } else {
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:         guard let first1 = first else {
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:           fatalError("Expected non-nil result 'first1' for nil error")
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:         }
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:         guard let second1 = second else {
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:           fatalError("Expected non-nil result 'second1' for nil error")
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:         }
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:         continuation.resume(returning: (first1, second1))
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:       }
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:     }
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT:   }
+// MULTIPLE-RETURN-VALUES-AND-ERROR-NEXT: }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR %s
+func testReturnNonOptionalValuesForResultAndError(completion: (Int?, Error?) -> Void) {
+  withoutAsyncAlternativeBecauseOfMismatchedCompletionHandlerName { result in
+    completion(1, MyError())
+  }
+}
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR:      func testReturnNonOptionalValuesForResultAndError() async throws -> Int {
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR-NEXT:   return try await withCheckedThrowingContinuation { continuation in 
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR-NEXT:     withoutAsyncAlternativeBecauseOfMismatchedCompletionHandlerName { result in
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR-NEXT:       continuation.resume(throwing: MyError())
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR-NEXT:     }
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR-NEXT:   }
+// NON-OPTIONAL-VALUE-FOR-RESULT-AND-ERROR-NEXT: }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=MIXED-OPTIONAL-AND-NON-OPTIONAL-RESULT %s
+func testMixedOptionalAnNonOptionaResults(completion: (Int?, String?, Error?) -> Void) {
+  withoutAsyncAlternativeThrowing { (theResult, error) in
+    completion(theResult, "hi", nil)
+  }
+}
+// MIXED-OPTIONAL-AND-NON-OPTIONAL-RESULT:      func testMixedOptionalAnNonOptionaResults() async throws -> (Int, String) {
+// MIXED-OPTIONAL-AND-NON-OPTIONAL-RESULT-NEXT:   return try await withCheckedThrowingContinuation { continuation in 
+// MIXED-OPTIONAL-AND-NON-OPTIONAL-RESULT-NEXT:     withoutAsyncAlternativeThrowing { (theResult, error) in
+// MIXED-OPTIONAL-AND-NON-OPTIONAL-RESULT-NEXT:       guard let theResult1 = theResult else {
+// MIXED-OPTIONAL-AND-NON-OPTIONAL-RESULT-NEXT:         fatalError("Expected non-nil result 'theResult1' for nil error")
```

Comment:
The message here is a bit odd since error is completely ignored... Not sure it's worth changing in this case though.

Reply:
What do you think about changing *all* messages to one of
- `Expected non-nil result '...' in the success case`
- `Expected non-nil result '...' in the non-error case`
- `Expected non-nil result '...' if no error is thrown`

I think even in the 
```swift
if let error = error {
  continuation.resume(throwing: error)
} else {
  guard let first1 = first else {
    fatalError("Expected non-nil result 'first1' for nil error")
  }
  continuation.resume(returning: first1)
}
```
it’s not entirely clear what `nil error` refers to.

Reply:
Thinking about it, I went with the `Expected non-nil result '...' in the non-error case` message. 

I also added a special case to omit the result name if it is "result". I.e. instead of `Expected non-nil result 'result' in non-error case`, we now emit `Expected non-nil result in non-error case`. If the result variable has any other name, we still emit it.